fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/connection/FHIRDbTenantDatasourceConnectionStrategy.java

```diff
@@ -0,0 +1,218 @@
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.config.FHIRConfiguration;
+import com.ibm.fhir.config.FHIRRequestContext;
+import com.ibm.fhir.config.PropertyGroup;
+import com.ibm.fhir.database.utils.model.DbType;
+import com.ibm.fhir.exception.FHIRException;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.jdbc.dao.api.FHIRDbDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.FHIRDbDAOImpl;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDBConnectException;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDataAccessException;
+
+
+/**
+ * Hides the logic behind obtaining a JDBC {@link Connection} from the DAO code.
+ * 
+ * Uses datasource and tenant ids configured in the fhir-server-config to
+ * map directly to a managed datasource. All managed datasources must be
+ * available when the server starts. This differs from the proxy
+ * datasource strategy {@link FHIRDbPropsConnectionStrategy} which supports
+ * dynamic (programmatic) definition of managed datasources.
+ * 
+ * @implNote Refactored from {@link FHIRDbDAOImpl}. Improves separation of
+ *           concerns by removing connection management code from the DAO
+ *           and injecting it as a strategy instead. This not only simplifies
+ *           things, but also makes it easier to implement new strategies,
+ *           such as using a JEE datasource directly instead of the FHIR
+ *           proxy datasource used here.
+ */
+public class FHIRDbTenantDatasourceConnectionStrategy extends FHIRDbConnectionStrategyBase {
+    private static final Logger log = Logger.getLogger(FHIRDbDAOImpl.class.getName());
+    private static final String CLASSNAME = FHIRDbDAOImpl.class.getName();
+
+    // number of nanoseconds in a millisecond
+    private static final long NANOMS = 1000000;
+    
+    // JNDI address of the (proxy) datasource
+    private final String datasourceBaseName = "jdbc/fhir_";
+
+    // Cache of datasources we've found
+    private final Map<String, DataSource> datasourceMap = new ConcurrentHashMap<>();
+    
+    // the flavor of the database we are configured to represent
+    private final FHIRDbFlavor flavor;
+
+    /**
+     * Public constructor. The proxy datasource must be present (registered in JNDI)
+     * at server startup.
+     * @throws FHIRPersistenceDBConnectException if the proxy datasource is not configured
+     */
+    public FHIRDbTenantDatasourceConnectionStrategy(TransactionSynchronizationRegistry trxSyncRegistry, Action newConnectionAction) throws FHIRException {
+        super(trxSyncRegistry, newConnectionAction);
+
+        // Find the base JNDI name of the datasource we want to use
+        try {
+//            this.datasourceBaseName =
+//                    FHIRConfiguration.getInstance().loadConfiguration().getStringProperty(
+//                        FHIRConfiguration.PROPERTY_JDBC_DATASOURCE_JNDINAME, FHIRDbDAO.FHIRDB_JNDI_NAME_DEFAULT);
```

Comment:
OK, what is this code block doing?  it seems unused

Reply:
This is a new datasource strategy Lee and I discussed. It will allow us to create simpler database configurations using datasources directly from JNDI instead of our more complex proxy configuration. I used it temporarily when I was debugging the setSchema issue. The strategy implementation works, but needs some extra support from the FHIR server config side of things to make it usable.

Reply:
let's take the code out or add a comment

Reply:
I'll add a comment

Reply:
I'm sorry I don't see the comment

Reply:
*           Currently not used - just needs some additional integration
 *           with the fhir-server-configuration to be supported.
lines 48-49

Reply:
thank you